crates/build/src/lib.rs

```diff
@@ -119,14 +126,56 @@ pub fn build_program(path: &str) {
 ///
 /// # Arguments
 ///
-/// * `path` - A string slice that holds the path to the program directory.
+/// * `path` - A path to the guest program directory
+///
 /// * `args` - A [`BuildArgs`] struct that contains various build configuration options.
 ///
 /// Set the `SP1_SKIP_PROGRAM_BUILD` environment variable to `true` to skip building the program.
-pub fn build_program_with_args(path: &str, args: BuildArgs) {
+///
+/// ## Note: Using this function without an absolute path is not recommended.
+/// Try using the `build_program_from_path!` macro instead.
+pub fn build_program_with_args(path: impl AsRef<Path>, args: BuildArgs) {
     build_program_internal(path, Some(args))
 }
 
+/// Builds the program with the given arguments if the program at path, or one of its dependencies,
+///
+/// ### Note:
+/// This function is only exposed to support the `build_program_from_path!` macro.
+/// It is not recommended to use this function directly.
+pub fn build_program_with_maybe_args(path: impl AsRef<Path>, args: Option<BuildArgs>) {
```

Comment:
i think we should avoid exposing this function if possible, though ngl i'm not really sure how we'd do it.